src/networking.c

```diff
@@ -2119,6 +2124,48 @@ void resetClient(client *c) {
     }
 }
 
+/* Initializes the shared query buffer to a new sds with the default capacity */
+void initSharedQueryBuf(void) {
+    thread_shared_qb = sdsnewlen(NULL, PROTO_IOBUF_LEN);
+    sdsclear(thread_shared_qb);
+}
+
+/* Resets the shared query buffer used by the given client.
+ * If any data remained in the buffer, the client will take ownership of the buffer 
+ * and a new empty buffer will be allocated for the shared buffer. */
+void resetSharedQueryBuf(client *c) {
+    if (c->querybuf != thread_shared_qb) return;
```

Comment:
i see all the called is checking the c->querybuf == thread_shared_qb, and the function name is resetSharedxxx, it seems like using assert would be better here if we want to avoid mistaken calls

Reply:
Changed.